src/bp/invoice.rs

```diff
@@ -0,0 +1,128 @@
+// LNP/BP Core Library implementing LNPBP specifications & standards
+// Written in 2020 by
+//     Dr. Maxim Orlovsky <orlovsky@pandoracore.com>
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the MIT License
+// along with this software.
+// If not, see <https://opensource.org/licenses/MIT>.
+
+use bitcoin::hashes::sha256d;
+use bitcoin::secp256k1::Signature;
+use bitcoin::Address;
+use miniscript::{descriptor::DescriptorPublicKey, Descriptor, Miniscript};
+use url::Url;
+
+use crate::bp::blind::OutpointHash;
+use crate::bp::chain::AssetId;
+use crate::bp::{Chain, HashLock, P2pNetworkId, Psbt, ScriptPubkeyFormat};
+use crate::lnp::{tlv, Features};
+use crate::secp256k1;
+
+#[derive(Tlv)]
+#[lnpbp_crate(crate)]
+pub enum FieldType {
+    #[tlv(type = 0x01)]
+    Payers,
+}
+
+#[derive(Lnp)]
+#[tlv_types(FieldType)]
+#[lnpbp_crate(crate)]
+pub struct Invoice {
+    network: P2pNetworkId,
+
+    /// List of beneficiary dests ordered in most desirable first order
+    beneficiaries: Vec<Beneficiary>,
+
+    /// Optional list of payers authored to pay
+    #[tlv(type = FieldType::Payers)]
+    payers: Vec<Payer>,
+    quantity: Quantity,
+    price: Option<AmountExt>,
```

Comment:
what is the meaning of 'quantity' and 'price' fields? How do they relate with the amount that a payer should pay to get the invoice fulfilled?

Reply:
The idea comes from Offers (BOLT 14 proposal) from Rusty Russel. Basically invoice creator specifies price per item, and you can invoice multiple items by paying multiple of the invoice amount